### PR TITLE
[KNX] Add BAOS support

### PR DESCRIPTION
**Pull request by petero-dk**
Upgrade the calimero dependency to 2.5 which uses java11 and fixes many many things.

### CODE REVIEW

**Comment by petero-dk** [comment]
For now it uses the 2.5-M1 release until the official 2.5 release.

https://search.maven.org/artifact/com.github.calimero/calimero-core/2.5-M1/jar

**Comment by petero-dk** [comment]
@wborn since calimero is now at 2.5 and now officially part of openhab, the baos changes are minor, and backwards compatible.

**Comment by lolodomo** [comment]
Is it now finished or still work in progress?

File: bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/client/SerialClient.java [hunk 1]
```diff
@@ -39,20 +40,31 @@
...
 
     @Override
     protected @NonNull KNXNetworkLink establishConnection() throws KNXException, InterruptedException {
         try {
             RXTXVersion.getVersion();
             logger.debug("Establishing connection to KNX bus through FT1.2 on serial port {}.", serialPort);
-            return new KNXNetworkLinkFT12(serialPort, new TPSettings());
+            KNXMediumSettings settings = new TPSettings();
+            if (useCEMI) {
+                return KNXNetworkLinkFT12.newCemiLink(serialPort, settings); // , deviceType);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/client/SerialClient.java:64]: Please suppress the ending comment

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by lolodomo** [comment]
Can you fix DCO check? 

The default translations should be updated.

**Comment by lolodomo** [comment]
@sjka ; any comment?

**Comment by J-N-K** [comment]
What is the deviceType for? It's not used as far as I can see.

**Comment by petero-dk** [comment]
The device type is an addition that is waiting for a change in calimero, that is also why it is commented out but there for future easy implementation - some BAOS devices handle commands differently, but support for that has not been implemented in Calimero yet.

I can completely remove it, if that is better?

File: bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/client/SerialClient.java (hunk 1 above)
Comment (petero-dk) [reply bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/client/SerialClient.java]: Done

**Comment by petero-dk** [comment]
Cleaned code and removed unused deviceType

File: bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/config/SerialBridgeConfiguration.java [hunk 2]
```diff
@@ -22,7 +22,13 @@
 
     private String serialPort;
 
+    private boolean useCEMI;
+
     public String getSerialPort() {
         return serialPort;
     }
+
+    public Boolean getUseCEMI() {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/config/SerialBridgeConfiguration.java:31]: boolean

**Review by lolodomo — CHANGES_REQUESTED** [review]
Please also add 2 lines in the default properties file, for the label and description of your new parameter.

**Comment by lolodomo** [comment]
And please fix build by running man spotless:apply

**Comment by petero-dk** [comment]
> Please also add 2 lines in the default properties file, for the label and description of your new parameter.

Where is that, I have added description and label in the serial.xml file

**Comment by petero-dk** [comment]
I see changes are already merged in another PR, so closing this - THANKS!

**Comment by lolodomo** [comment]
No there was no other KNX PR merged in the openHAB repo.

**Pull request closed without merging**